atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/DisruptorAutobatcher.java

```diff
@@ -85,6 +89,22 @@ private static ThreadFactory createThreadFactory(String safeLoggablePurpose) {
         return result;
     }
 
+    public List<ListenableFuture<R>> applyBatch(List<T> arguments) {
+        Preconditions.checkState(!closed, "Autobatcher is already shut down");
+        List<ListenableFuture<R>> results = new ArrayList<>();
+        buffer.publishEvents((EventTranslator<DefaultBatchElement<T, R>>[]) arguments.stream().map(argument -> {
```

Comment:
I think it is possible that `buffer.publishEvents` does not call every one of its arguments; this is the only case that I can see where we would end up with less transactions than expected (in the sense that, elsewhere, we will have the right number of `Optional<Thing>`. It is possible that we may want it to return `Optionals` at even this level - i.e. if `results.size() < arguments.size()` or something akin to that.

Reply:
It should be pretty clear from documentation what this method does, having to return Optionals here would be wrong API for us.

Reply:
I checked, and it does call. If you wanted it to fail if it can't do the whole batch, there's try* version of this method.